apps/explorer/lib/explorer/chain/celo/celo_election_rewards.ex

```diff
@@ -70,6 +73,57 @@ defmodule Explorer.Chain.CeloElectionRewards do
     )
   end
 
+  def base_aggregated_block_query(block_number, reward_types) do
+    {:ok, zero_wei} = Wei.cast(0)
+
+    from(rewards in __MODULE__,
+      group_by: rewards.block_number,
+      group_by: rewards.reward_type,
+      where: rewards.amount != ^zero_wei,
+      where: rewards.block_number == ^block_number,
+      where: rewards.reward_type in ^reward_types
+    )
+  end
+
+  def aggregated_voter_and_validator_query(block_number) do
+    query = base_aggregated_block_query(block_number, ["validator", "voter"])
+
+    query
+    |> select([rewards], %{
+      reward_type: rewards.reward_type,
+      amount: sum(rewards.amount),
+      count: count()
+    })
+  end
+
+  def aggregated_validator_group_query(block_number) do
+    query = base_aggregated_block_query(block_number, ["group"])
+
+    query
+    |> select([rewards], %{
+      reward_type: rewards.reward_type,
+      amount: sum(rewards.amount),
+      validator_count: fragment("COUNT(DISTINCT(associated_account_hash))"),
+      group_count: fragment("COUNT(DISTINCT(account_hash))"),
+      count: count()
+    })
+  end
+
+  def get_aggregated_for_block_number(block_number) do
+    default = %{voter: nil, group: nil, validator: nil}
+    validator_group_query = aggregated_validator_group_query(block_number)
+    voter_and_validator_query = aggregated_voter_and_validator_query(block_number)
+
+    default
+    |> Map.merge(
+      voter_and_validator_query
```

Comment:
lets move these into an intermediate variables

it adds more context around why operations are happening, and then instead of `Enum.map |> Map.new |> Map.put` we can do 

`Enum.into(%{group: validator_group_aggregates}, fn reward -> {String.to_exisiting_atom(reward.reward_type), rewards} end)`

Reply:
I couldn't get it to work with `Enum.into` anyhow, refactored it to be more verbose though, please let me know if that looks better to you

Reply:
It's not about verbosity, it's about clarity - without context there's no reason why these structures need to be manipulated in such a way. Transforming the data at the same time as fetching the data makes it harder to reason about the structure.

with something like

```elixir
#...

validator_group_rewards = validator_group_query |> Repo.one()

default = %{voter: nil, group: validator_group_rewards, validator: nil}

voter_and_validator_query
|> Repo.all()
|> Enum.into(default, fn rewards -> {String.to_existing_atom(rewards.reward_type), rewards} end)
```

we first fetch the data and then do all the transformation in the last step of the pipeline